lib/auth/auth_with_roles.go

```diff
@@ -6111,7 +6157,7 @@ func (a *ServerWithRoles) UpdateClusterMaintenanceConfig(ctx context.Context, cm
 
 	if modules.GetModules().Features().Cloud {
 		// maintenance configuration in cloud is derived from values stored in
-		// an external cloud-specific databse.
+		// an external cloud-specific database.
 		return trace.NotImplemented("cloud clusters not support custom cluster maintenance resources")
```

Comment:
```suggestion
		return trace.NotImplemented("cloud clusters do not support custom cluster maintenance resources")
```

while we're here 🤷‍♂️